solidity/test/TestPhasedEscrow.js

```diff
@@ -426,7 +425,378 @@ describe("PhasedEscrow", () => {
+    await phasedEscrow.withdraw(amount, {from: owner})
+
+    // Verify that funds got transferred to BatchedPhasedEscrow
+    expect(await token.balanceOf(batchedPhasedEscrow.address)).to.eq.BN(
+      amount,
+      `Unexpected batched escrow balance`
+    )
+    expect(await token.balanceOf(phasedEscrow.address)).to.eq.BN(
+      0,
+      `Unexpected phased escrow balance`
+    )
+  })
+
+  describe("receiveApproval", async () => {
+    it("fails for an unknown token", async () => {
+      // It is another KeepToken contract deployment, not the one PhasedEscrow
+      // has been created with.
+      const unknownToken = await KeepToken.new({from: owner})
+      const amountApproved = web3.utils.toBN(9991)
+
+      await expectRevert(
+        unknownToken.approveAndCall(
+          batchedPhasedEscrow.address,
+          amountApproved,
+          "0x0",
+          {from: owner}
+        ),
+        "Unsupported token"
+      )
+    })
+
+    it("transfers all approved tokens", async () => {
+      const amountApproved = web3.utils.toBN(9993)
+      await token.approveAndCall(
+        batchedPhasedEscrow.address,
+        amountApproved,
+        "0x0",
+        {from: owner}
+      )
+
+      const actualBalance = await token.balanceOf(batchedPhasedEscrow.address)
+      expect(actualBalance).to.eq.BN(amountApproved)
+    })
+  })
+
+  describe("beneficiary approval", async () => {
+    it("can be done by owner", async () => {
+      await batchedPhasedEscrow.approveBeneficiary(beneficiary1.address, {
+        from: owner,
+      })
+      // ok, no revert
+    })
+
+    it("can be done by updated owner", async () => {
+      await batchedPhasedEscrow.transferOwnership(updatedOwner, {from: owner})
+
+      await expectRevert(
+        batchedPhasedEscrow.approveBeneficiary(beneficiary1.address, {
+          from: owner,
+        }),
+        "Ownable: caller is not the owner"
+      )
+      await batchedPhasedEscrow.approveBeneficiary(beneficiary1.address, {
+        from: updatedOwner,
+      })
+      // ok, no revert
+    })
+
+    it("can not be done by non-owner", async () => {
+      await expectRevert(
+        batchedPhasedEscrow.approveBeneficiary(beneficiary1.address, {
+          from: drawee,
+        }),
+        "Ownable: caller is not the owner"
+      )
+    })
+
+    it("can not be done on zero address", async () => {
+      await expectRevert(
+        batchedPhasedEscrow.approveBeneficiary(ZERO_ADDRESS, {from: owner}),
+        "Beneficiary can not be zero address"
+      )
+    })
+
+    it("emits an event", async () => {
+      const receipt = await batchedPhasedEscrow.approveBeneficiary(
+        beneficiary1.address,
+        {
+          from: owner,
+        }
+      )
+
+      expectEvent(receipt, "BeneficiaryApproved", {
+        beneficiary: beneficiary1.address,
+      })
+    })
+
+    it("maintains beneficiaries as non-approved by default", async () => {
```

Comment:
Idk, but it feels like we're testing the default Solidity values here, which for `bool` is always `false`

Reply:
Yeah, could be. Though I was looking at `BatchedPhasedEscrow` as a black-box, without thinking how this mechanism is implemented. This could be, for example, a list of approved beneficiaries, with some implementation bug that would return `true` in case that list would be empty.

Reply:
For sure it wouldn't hurt to have it. Okay, let's leave it